src/Aspire.Hosting.Azure.PostgreSQL/AzurePostgresFlexibleServerResource.cs

```diff
@@ -98,4 +104,27 @@ internal ReferenceExpression GetDatabaseConnectionString(string databaseResource
     }
 
     internal static string GetDatabaseKeyVaultSecretName(string databaseResourceName) => $"{databaseResourceName}-connectionString";
+
+    /// <inheritdoc/>
+    public override ProvisionableResource AddAsExistingResource(AzureResourceInfrastructure infra)
+    {
+        var store = PostgreSqlFlexibleServer.FromExisting(this.GetBicepIdentifier());
+        store.Name = NameOutputReference.AsProvisioningParameter(infra);
+        infra.Add(store);
+        return store;
+    }
+
+    /// <inheritdoc/>
+    public override void AddRoleAssignments(AddRoleAssignmentsContext roleAssignmentContext)
+    {
+        Debug.Assert(!UsePasswordAuthentication, "AddRoleAssignments should not be called when using UsePasswordAuthentication");
+
+        var infra = roleAssignmentContext.Infrastructure;
+        var postgres = (PostgreSqlFlexibleServer)AddAsExistingResource(infra);
+
+        var principalId = roleAssignmentContext.PrincipalId;
+        var principalName = roleAssignmentContext.PrincipalName;
+
+        AzurePostgresExtensions.AddActiveDirectoryAdministrator(infra, postgres, principalId, PostgreSqlFlexibleServerPrincipalType.ServicePrincipal, principalName);
```

Comment:
Instead of hard-coding `PostgreSqlFlexibleServerPrincipalType.ServicePrincipal` here, we should get the context's PrinciaplType and then translate it into a `PostgreSqlFlexibleServerPrincipalType`